stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -174,28 +200,48 @@ extension Unsafe${Mutable}BufferPointer: Sequence {
     d.initialize(from: s, count: n)
     return (Iterator(_position: s + n, _end: s + count), n)
   }
+
+  @inlinable
+  public func withContiguousStorageIfAvailable<R>(
```

Comment:
Do we want to make this available `where Pointee: ~Copyable` like you did for a lot of the collection API?

Reply:
Although this is only really useful in generic cases where you know something is a `Sequence` so maybe not.

Reply:
The `Iterator` requires `Pointee` to be copyable, so this cannot do more.

The copyability of `R` ought to get relaxed somehow, but I don't want to mess with it just yet -- and this whole API will get replaced by spans/storage views, so we may never actually do it.

Reply:
We do have to mess with typed errors for this whole API; we can mess with the return value at that time.